src/main/java/com/example/algorithm/QVoterModel.java

```diff
@@ -86,19 +84,20 @@ private StepReport makeDecision() {
         report.fillRoles(selectedAgent, null);
 
         if (shouldAcceptNeighboursOpinion()) {
-            selectedAgent.setIsSupporting(opinionsReceived.get(0));
+            selectedAgent.setIsSupporting(selectedAgent.getMajorityVote());
```

Comment:
Are those votes cleaned after each voting? If they aren't, are You sure that voter should get majority from his knowledge table? As far as I understand this algorithm, we takes into account only votes he gets from neighbours at each step.